### PR TITLE
Feat/#40011

### PR DESCRIPTION
**Pull request by SDmayank**

### CODE REVIEW

File: src/repositories/user/IUserModel.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as mongoose from 'mongoose'
```
Comment (surajSS12) [inline src/repositories/user/IUserModel.ts]: add semicolon
always give a blank line after last import

File: src/repositories/user/UserSchema.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+emails: String,
+dob: Date,
+mobileNumber: Number,
+hobbies: [String]
+};
+super(userSchema, options );
+}
+}
+
+export default UserSchema;
```
Comment (surajSS12) [inline src/repositories/user/UserSchema.ts]: code not intended

File: src/repositories/user/UserSchema.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
+import * as mongoose from 'mongoose';
+
+class UserSchema extends mongoose.Schema {
+constructor( options) {
```
Comment (surajSS12) [inline src/repositories/user/UserSchema.ts]: extra space after '{'

File: src/repositories/user/UserRepository.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { userModel } from './UserModel';
+import * as mongoose from 'mongoose';
+import IUserModel from './IUserModel';
+import IUserCreate from './entities/IUserCreate';
+
+class UserRepository {
+    private userModel: mongoose.Model<IUserModel>
```
Comment (surajSS12) [inline src/repositories/user/UserRepository.ts]: Add semicolon

File: src/repositories/user/UserModel.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import * as mongoose from 'mongoose';
+import IUserModel from './IUserModel';
+import UserSchema from './UserSchema';
+
+export const userSchema = new UserSchema({
+    collection: 'Users'
+});
+
+export const userModel: mongoose.Model<IUserModel> = mongoose.model<IUserModel>('user', userSchema, 'Users',true);
```
Comment (surajSS12) [inline src/repositories/user/UserModel.ts]: No whitespace between 'Users' and true

File: src/repositories/user/UserModel.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import * as mongoose from 'mongoose';
+import IUserModel from './IUserModel';
+import UserSchema from './UserSchema';
+
+export const userSchema = new UserSchema({
+    collection: 'Users'
```
Comment (surajSS12) [inline src/repositories/user/UserModel.ts:6]: Extra tab

File: src/repositories/user/IUserModel.ts (hunk 1 above)
Comment (SDmayank) [reply src/repositories/user/IUserModel.ts]: done

File: src/repositories/user/UserSchema.ts (hunk 2 above)
Comment (SDmayank) [reply src/repositories/user/UserSchema.ts]: done

File: src/repositories/user/UserSchema.ts (hunk 3 above)
Comment (SDmayank) [reply src/repositories/user/UserSchema.ts]: done

File: src/repositories/user/UserRepository.ts (hunk 4 above)
Comment (SDmayank) [reply src/repositories/user/UserRepository.ts]: done

File: src/repositories/user/UserModel.ts (hunk 5 above)
Comment (SDmayank) [reply src/repositories/user/UserModel.ts]: done

File: src/repositories/user/UserModel.ts (hunk 6 above)
Comment (SDmayank) [reply src/repositories/user/UserModel.ts:6]: done

**Pull request merged**